### PR TITLE
[stable/telegraf] Check for service.enabled

### LINKED ISSUE #15262: [stable/telegraf] spec.ports: Required value
Using this values, doesn't work:

```
env:
- name: "AZURE_CLIENT_ID"
  value: "XXXX"
- name: "AZURE_CLIENT_SECRET"
  value: "XXXX"
- name: "AZURE_TENANT_ID"
  value: "XXXX"

config:
  agent:
    debug: true
    hostname: "telegraf"
  outputs:
    - azure_monitor:
        region: "eastus"
        resource_id: "/subscriptions/XXXX/resourceGroups/XXXXX/providers/XXXXXX/workspaces/XXXXXX"
        strings_as_dimensions: true
  inputs:
    - rabbitmq:
        url: "http://XXXXXXXX:15672"
        username: "XXXXX"
        password: "XXXXXXX"
```

**Describe the bug**
```
helm install --name telegraf stable/telegraf -f .\values.yaml
Error: release telegraf failed: Service "telegraf" is invalid: spec.ports: Required value
```
**Version of Helm and Kubernetes**:
```
helm version
Client: &version.Version{SemVer:"v2.13.1", GitCommit:"618447cbf203d147601b4b9bd7f8c37a5d39fbb4", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.13.1", itCommit:"618447cbf203d147601b4b9bd7f8c37a5d39fbb4", GitTreeState:"clean"}

kubectl version
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.11", GitCommit:"637c7e288581ee40ab4ca210618a89a555b6e7e9", GitTreeState:"clean", BuildDate:"2018-11-26T14:38:32Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"windows/amd64"}
Server Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.3", GitCommit:"a4529464e4629c21224b3d52edfe0ea91b072862", GitTreeState:"clean", BuildDate:"2018-09-09T17:53:03Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}

```

**Which chart**:
stable/telegraf

### PR DESCRIPTION
**Pull request by raphaelquati**
#### What this PR does / why we need it:
Check for service.enabled

#### Which issue this PR fixes
  - fixes #15262

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [X] Chart Version bumped
- [X] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

**Comment by naseemkullah** [comment]
Hi @raphaelquati ,

 It seems like port of service will depend on the `config.inputs` values

 Instead of adding an option to disable the service, keep it and please configure a similar logic for `rabbitmq` here:
https://github.com/helm/charts/blob/b91e719e6812a15bec9afcd9f0b6e7554cb564ba/stable/telegraf/templates/service.yaml#L12-L42

 ?

 I _think_ that should work.
 Cheers!

**Comment by naseemkullah** [comment]
I think you will need `{{ trimPrefix ":" $value.url | int64 }}` instead of `{{ trimPrefix ":" $value.service_address | int64 }}` like the ones that are there

**Comment by raphaelquati** [comment]
I think the purpose of the service ports is intended to plugins make Telegraph listen external ports (like https://github.com/influxdata/telegraf/tree/master/plugins/inputs/socket_listener or https://github.com/influxdata/telegraf/tree/master/plugins/inputs/influxdb_listener). In the case of url in rabbitmq example, is the address of RabbitMQ server, and not make Telegraph listen that address.

File: stable/telegraf/templates/service.yaml [hunk 1]
```diff
@@ -1,3 +1,5 @@
+{{- if .Values.service.enabled -}}
+{{- if .Values.service.enabled -}}
```
Comment (naseemkullah) [inline stable/telegraf/templates/service.yaml]: Can you remove one of these 2?

**Comment by naseemkullah** [comment]
Thanks for the explanation.

**Review by naseemkullah — APPROVED** [review]

**Comment by naseemkullah** [comment]
Needs a chart version bump according to CI test result @raphaelquati

**Pull request merged**